### PR TITLE
Add support for Nginx 1.21.4

### PR DESCRIPTION
**Pull request by julek-wolfssl**
- Add KEYGEN to Nginx config
- Check for name length in `wolfSSL_X509_get_subject_name`
- Refactor `wolfSSL_CONF_cmd`
- Implement `wolfSSL_CONF_cmd_value_type`
- Don't forecfully overwrite side

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -57761,80 +57761,77 @@ static const conf_cmd_tbl conf_cmds_tbl[] = {
...
- * @param cmd   configuration command
- * @param value arguments for cmd
- * @return 1 when cmd is recognised, but value is not used
- *         2 both cmd and value are used
- *  otherwise WOLFSSL_FAILURE
- *         -2 if cmd is not recognised
- *         -3 if value is NULL, but cmd is recognized
- */
-int wolfSSL_CONF_cmd(WOLFSSL_CONF_CTX* cctx, const char* cmd, const char* value)
+static const conf_cmd_tbl* CONF_find_cmd(WOLFSSL_CONF_CTX* cctx,
```
Comment (SparkiDev) [inline src/ssl.c:57764]: This API looks like the OpenSSL one now. Rename to wolfssl_conf_find_cmd() to make it clear.

File: tests/api.c [hunk 2]
```diff
@@ -32747,7 +32747,7 @@ static void test_wolfSSL_X509_STORE_CTX_get0_current_issuer(void)
         cmp = X509_NAME_cmp(caName, issuerName);
         AssertIntEQ(cmp, 0);
     #else
-        AssertNotNull(issuerName);
+        AssertNull(issuerName);
```
Comment (SparkiDev) [inline tests/api.c:32750]: Issuer name is not NULL for me when I compile with --enable-opensslextra only.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/ssl.c:57764]: Renamed.

Comment (SparkiDev) [reply src/ssl.c:57764]: Sorry, but the name hasn't changed in the pull request. Did you push up the change?

File: tests/api.c (hunk 2 above)
Comment (SparkiDev) [reply tests/api.c:32750]: Ah! OK.

File: src/ssl.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/ssl.c:57764]: D'oh! Forgot to push the changes. Pushed now so you should see the rename.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**